src/solver/optimisation/opt_alloc_probleme_a_optimiser.cpp

```diff
@@ -216,22 +209,19 @@ static void optimisationAllocateProblem(PROBLEME_HEBDO* problemeHebdo, const int
 
 void OPT_AllocDuProblemeAOptimiser(PROBLEME_HEBDO* problemeHebdo)
 {
-    int mxPaliers;
-    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre;
-    ProblemeAResoudre
-      = (PROBLEME_ANTARES_A_RESOUDRE*)MemAllocMemset(sizeof(PROBLEME_ANTARES_A_RESOUDRE));
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre
+        = (PROBLEME_ANTARES_A_RESOUDRE*)MemAllocMemset(sizeof(PROBLEME_ANTARES_A_RESOUDRE));
     problemeHebdo->ProblemeAResoudre = ProblemeAResoudre;
 
+    int mxPaliers;
     OPT_DecompteDesVariablesEtDesContraintesDuProblemeAOptimiser(problemeHebdo, &mxPaliers);
```

Comment:
```suggestion
    int mxPaliers = OPT_DecompteDesVariablesEtDesContraintesDuProblemeAOptimiser(problemeHebdo);
```
This requires a change in function `OPT_DecompteDesVariablesEtDesContraintesDuProblemeAOptimiser`